### DISCUSSION TITLE
help with modules

### DISCUSSION THREAD

**Discussion by LollipopFt** [orbitalquark/textadept#203] [Q&A] [answered]
hi, I have this module with filename `lineguides.lua`:
```lua
for _, i in ipairs(_G.guides) do
    _G.view:multi_edge_add_line(i, 0x808080)
end

_G.view.edge_mode = view.EDGE_MULTILINE
```
and have this in my `init.lua`:
```lua
_G.guides = {80, 100}
```
to get line guides at the specified columns. however, i see other plugins using something like
```lua
local M = {}
M.guides = {}
```
and their `init.lua` file with
```lua
[module name].guides = {80, 100)
```
and getting values from there. i have tried using that but failed so may i know how this can be achieved since i do not want to use global(`_G`) variables? thank you!

**Comment by rgieseke** [comment, marked as answer]
I think the following should work:

`lineguide.lua`
```lua
local M = {}

M.guides = {}

events.connect(events.INITIALIZED, function()
  for _, i in ipairs(M.guides) do
      view:multi_edge_add_line(i, 0x808080)
  end

  view.edge_mode = view.EDGE_MULTILINE
end)

return M
```

`init.lua`
```lua
local lineguide = require("lineguide")
lineguide.guides = {80, 100}
```

The `return M` makes the modules variables available when you import it. Since the `guides` variable is not set when you require the module the edge setting is postponed until the initialization finishes.

**Comment by LollipopFt** [comment]
i see, thank you!
